src/main/java/com/minecolonies/coremod/research/LocalResearchTree.java

```diff
@@ -94,7 +129,68 @@ public boolean branchFinishedHighestLevel(final String branch)
     @Override
     public void finishResearch(final String id)
     {
-        inProgress.remove(id);
+        inProgress.remove(id); isComplete.add(id);
+    }
+
+    @Override
+    public void cancelResearch(final String branch, final String id, final IColony colony)
+    {
+        checkAndResetDescendants(branch, Arrays.asList(id));
```

Comment:
I would only allow to reset research without concluded parents. This way players have to reset one by one.